plugins/updater/src/updater.rs

```diff
@@ -587,6 +587,27 @@ impl Update {
         msi_path.push(&path);
         msi_path.push("\"");
 
+        let mut msi_args = Vec::new();
+        for arg in self.nsis_installer_args()[2..].iter() {
+            let arg = arg.to_string_lossy();
+
+            if !arg.contains(' ') {
+                msi_args.push(arg.to_string());
+                continue;
+            }
+
+            if arg.starts_with('-') {
+                if let Some((a1, a2)) = arg.split_once('=') {
+                    msi_args.push(format!("{a1}=\"\"{a2}\"\""));
+                } else {
+                    msi_args.push(format!("\"\"{arg}\"\""));
+                }
+            } else {
+                msi_args.push(format!("\"\"{arg}\"\""));
+            }
```

Comment:
this is some wild code, let's move it into a separate function and add extensive unit tests for it

Reply:
The fun stuff for the tests will be to figure out what's actually the intended result lol

Reply:
left a comment below